### PR TITLE
UF-5123: Upgraded to SB3

### PR DESCRIPTION
**Pull request by MickeWallner**

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -15,8 +15,8 @@ openapi.version=@project.version@
 # Datasource settings
 #========================================
 # DDL script generation
-spring.jpa.properties.javax.persistence.schema-generation.database.action=none
+spring.jpa.properties.jakarta.persistence.schema-generation.database.action=none
 # Drivers, dialect, etc.
-spring.jpa.database-platform=org.hibernate.dialect.MariaDB106Dialect
+spring.jpa.database-platform=org.hibernate.dialect.MariaDBDialect
```
Comment (foobar-1337) [inline src/main/resources/application.properties:20]: Can also be removed

File: src/test/java/se/sundsvall/billingpreprocessor/integration/db/BillingRecordRepositoryTest.java [hunk 2]
```diff
@@ -43,6 +22,24 @@
...
+import static se.sundsvall.billingpreprocessor.api.model.enums.Status.APPROVED;
+import static se.sundsvall.billingpreprocessor.api.model.enums.Status.NEW;
+import static se.sundsvall.billingpreprocessor.api.model.enums.Status.REJECTED;
+import static se.sundsvall.billingpreprocessor.api.model.enums.Type.EXTERNAL;
+import static se.sundsvall.billingpreprocessor.integration.db.model.DescriptionType.DETAILED;
+import static se.sundsvall.billingpreprocessor.integration.db.model.DescriptionType.STANDARD;
+
 /**
  * billingRecord repository tests
  *
```
Comment (foobar-1337) [inline src/test/java/se/sundsvall/billingpreprocessor/integration/db/BillingRecordRepositoryTest.java:45]: `@DataJpaTest` instead?

File: src/test/java/se/sundsvall/billingpreprocessor/integration/db/SchemaVerificationTest.java [hunk 3]
```diff
@@ -1,25 +1,25 @@
...
 import java.nio.file.Path;
 import java.nio.file.Paths;
 
-import org.junit.jupiter.api.Test;
-import org.springframework.beans.factory.annotation.Value;
-import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.test.context.ActiveProfiles;
+import static org.assertj.core.api.Assertions.assertThat;
 
 @SpringBootTest
```
Comment (foobar-1337) [inline src/test/java/se/sundsvall/billingpreprocessor/integration/db/SchemaVerificationTest.java:16]: `@DataJpaTest` instead?

File: src/main/java/se/sundsvall/billingpreprocessor/integration/db/BillingRecordRepository.java [hunk 4]
```diff
@@ -1,11 +1,9 @@
 package se.sundsvall.billingpreprocessor.integration.db;
 
-import javax.transaction.Transactional;
-
+import io.github.resilience4j.circuitbreaker.annotation.CircuitBreaker;
+import jakarta.transaction.Transactional;
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/billingpreprocessor/integration/db/BillingRecordRepository.java:4]: Isn't `@org.springframework.transaction.annotation.Transactional` preferred in SpringBoot-applications?

File: src/test/java/se/sundsvall/billingpreprocessor/integration/db/BillingRecordRepositoryTest.java [hunk 5]
```diff
@@ -1,25 +1,7 @@
...
-
-import java.time.LocalDate;
-import java.time.OffsetDateTime;
-import java.util.ArrayList;
-import java.util.List;
-
-import javax.transaction.Transactional;
-
+import com.turkraft.springfilter.converter.FilterSpecificationConverter;
+import jakarta.transaction.Transactional;
```
Comment (foobar-1337) [inline src/test/java/se/sundsvall/billingpreprocessor/integration/db/BillingRecordRepositoryTest.java:4]: Isn't `@org.springframework.transaction.annotation.Transactional` preferred in SpringBoot-applications?

**Comment by foobar-1337** [comment]
You also need to change som javax-properties in the properties repo as well.

File: src/test/java/se/sundsvall/billingpreprocessor/integration/db/BillingRecordRepositoryTest.java (hunk 2 above)
Comment (MickeWallner) [reply src/test/java/se/sundsvall/billingpreprocessor/integration/db/BillingRecordRepositoryTest.java:45]: Then FilterSpecificationConverter is not availiable.

**Review by foobar-1337 — APPROVED** [review]

**Pull request merged**